### ISSUE TITLE
Add coordinates from NDCube sliced out axes to GlobalCoords

### ISSUE THREAD

**Issue by DanRyanIrish** [sunpy/ndcube#334]
### Description

#323 creates a new `GlobalCoords` class.  Now, coordinates corresponding axes sliced out of an `NDCube` (i.e. when slicing reduces the dimensionality of an `NDCube`) must be added to the instance of the `GlobalCoords` attached to the `NDCube`.  Therefore the following is needed:
* ~An API for initialising an `NDCube` with `GlobalCoords` must be created.~
* When an axis is dropped as part of slicing an `NDCube`, the value of each coordinate at the point along the axis where the cube was sliced must be transferred to `GlobalCoords`.  If that coordinate still depends on another axis that has not been sliced away, then it should not be added to `GlobalCoords`.  This should be implemented in `NDCube`'s `__getitem__`.
* The above must be done for coords from both the `wcs` and the `extra_coords`.
* Testing the right coords have been inserted when slicing occurs must be added.

**Comment by Cadair** [comment]
To add a little more detail on how this should be implemented. In astropy 4.2 `SlicedLowLevelWCS` has a [`dropped_world_dimensions`](https://github.com/astropy/astropy/blob/master/astropy/wcs/wcsapi/wrappers/sliced_wcs.py#L157) method which contains all the information needed to generate the elements of the `GlobalCoords` object. This property should be accessed inside `GlobalCoords._all_coords` and the low level objects converted to hight level objects with code similar to `HighLevelWCSWrapper.pixel_to_world`.

I also propose we implement the same system in `ExtraCoords` where when `ExtraCoords.__getitem__` is called a `ExtraCoords.dropped_world_dimensions` property is populated, returning the same information as the `SlicedLowLevel` equivalent.

**Comment by DanRyanIrish** [comment]
:+1:

**Comment by DanRyanIrish** [comment]
@rosteen, you might be interested in tackling this issue.  To start with I would just tackle the case of adding sliced out global coordinates from the WCS object and leave the extra_coords case for a later PR.

**Comment by Cadair** [comment]
This is on the top of my list, but I am happy to tackle the ExtraCoords part if someone else wants to take a swing at the wcs part.

**Comment by DanRyanIrish** [comment]
Splitting this into two PRs might not be the worst idea anyway.  So why don't you, @Cadair, start with the `ExtraCoords` part to give @rosteen a chance to do the WCS part.  The time saved might also allow you to look at one of the other outstanding `ExtraCoords` issues as well?

**Comment by rosteen** [comment]
I am mostly unavailable for `ndcube` work for the first few days of this week, so if @Cadair can knock it all out don't feel the need to split it up and wait on me. Otherwise I can take a look later in the week at the WCS stuff (I do need to get more familiar with WCS in general).

**Comment by Cadair** [comment]
@rosteen You take the WCS part, hopefully [my comment](https://github.com/sunpy/ndcube/issues/334#issuecomment-728089461) has enough to get you started, but feel free to ping me in chat or on here if you get stuck.

**Comment by rosteen** [comment]
Alright, after poking around with this I think I understand how to implement, but I need a clarification of what exactly should end up in `GlobalCoords`. Say we we take a slice [0, :, :] out of an `NDCube` of shape (1000, 20, 20), where the first axis is the wavelength and the other two are spatial. Is the thing we want to put into the attached `GlobalCoords` a shape (20, 20) array that contains wavelength value 0 at every location? Or something else?

One more question - it looks like `GlobalCoords` stores a dictionary keyed on coordinate name - if the original `wcs` had no `world_axis_names` value specified, should be just make one up? Perhaps "dropped_spectral_1" or something, generated based on coordinate type.

**Comment by rosteen** [comment]
Rereading your comment, sounds like maybe the answer is we don't store an array at all but rather generate it on the fly whenever `GlobalCoords._all_coords` is called, which seems to remove the need for any code additions to `NDCube`'s `__getitem__` for at least the `wcs` part of this (the check for any `dropped_world_coordinates` can just live in `GlobalCoords._all_coords`).

**Comment by DanRyanIrish** [comment]
> Alright, after poking around with this I think I understand how to implement, but I need a clarification of what exactly should end up in `GlobalCoords`. Say we we take a slice [0, :, :] out of an `NDCube` of shape (1000, 20, 20), where the first axis is the wavelength and the other two are spatial. Is the thing we want to put into the attached `GlobalCoords` a shape (20, 20) array that contains wavelength value 0 at every location? Or something else?
> 
> One more question - it looks like `GlobalCoords` stores a dictionary keyed on coordinate name - if the original `wcs` had no `world_axis_names` value specified, should we just make one up? Perhaps "dropped_spectral_1" or something, generated based on coordinate type.

**Comment by DanRyanIrish** [comment]
Another thing to mention is that we should only add coords to global_coords if all array axes to which it corresponds are slices away. So in your above example, if you slice the cube with [:, 0, :], no spatial coord should be added. But if you use [:, 0, 0], then both lat and lon spatial coords should be added.
